### PR TITLE
Added api for removing dismiss listener.

### PR DESCRIPTION
**Pull request by eldhosembabu**
Added api for removing dismiss listener.

Motivational PR: https://github.com/firebase/firebase-android-sdk/pull/3980

Approved API Review: b/262402247

### CODE REVIEW

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/FirebaseInAppMessaging.java [hunk 1]
```diff
@@ -336,6 +336,16 @@ public void removeDisplayErrorListener(
     developerListenerManager.removeDisplayErrorListener(displayErrorListener);
   }
 
+  /**
+   * Unregisters a dismiss listener.
+   *
+   * @param dismissListener
```
Comment (rlazo) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/FirebaseInAppMessaging.java:342]: Would more info about the param be helpful?

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DeveloperListenerManager.java [hunk 2]
```diff
@@ -145,10 +145,15 @@ public void removeDisplayErrorListener(
...
 
+  public void removeDismissListener(FirebaseInAppMessagingDismissListener dismissListener) {
+    registeredDismissListeners.remove(dismissListener);
+  }
+
   public void removeAllListeners() {
     registeredClickListeners.clear();
     registeredImpressionListeners.clear();
     registeredErrorListeners.clear();
+    registeredDismissListeners.clear();
```
Comment (rlazo) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DeveloperListenerManager.java:156]: Do you have any tests for this code? If so, adding an extra check for the dismiss listener sounds reasonable.

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/FirebaseInAppMessaging.java (hunk 1 above)
Comment (eldhosembabu) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/FirebaseInAppMessaging.java]: Done

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DeveloperListenerManager.java (hunk 2 above)
Comment (eldhosembabu) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DeveloperListenerManager.java:157]: Added new tests to check the entire add remove listeners functionality and methods.

**Comment by rlazo** [comment]
Don't forget to update your `api.txt` file running

```
./gradlew firebase-inappmessaging:generateApiTxtFile
```

and update the SDK `version` in https://github.com/firebase/firebase-android-sdk/tree/master/firebase-inappmessaging/gradle.properties if necessary.

**Comment by eldhosembabu** [comment]
Done..Please check the version update is correct. Thanks.

**Review by rlazo — APPROVED** [review]

**Pull request merged**